### PR TITLE
feat(shareable-lists): paginated Items on a List

### PR DESCRIPTION
**Pull request by kschelonka**
Provides pagination API for items on a list, deprecating the old API which returned all Items in the list in a single response.

This is fine for the majority of cases, but we already see users with thousands of items in their lists _even when they have to do it one-at-a-time_. Once the bulk api is enabled in the frontend, we can probably anticipate that list sizes will grow.

**Changes**
* Update the pagination validation util -- make the type responses more specific/accurate, and optionally disable the cursor validator
* Add a paginated field `items` to ShareableListItem, which implements the Relay cursor pagination spec; deprecate `listItems` field
* Fake cursor pagination in the backend (decode into offset pagination)

**Not Changed**
* the Admin API - my reasoning is we're not allowing public lists to be created, and we'll update the API if/when we need to have administrative access

### CODE REVIEW

File: servers/shareable-lists-api/src/database/mutations/ShareableList.ts [hunk 1]
```diff
@@ -173,6 +173,7 @@ export async function createAndAddToShareableList(
   const completeList: ShareableList = {
     ...list,
     // bigint causes issues again...
+    id: list.id as unknown as bigint,
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/mutations/ShareableList.ts:176]: 🤣

File: servers/shareable-lists-api/src/models/ShareableListItem.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  constructor(private context: BaseContext) {}
+  byListId(
+    listId: number,
+  ): SelectQueryBuilder<DB, 'ListItem', ListItemResponse> {
+    const qb = this.context.conn
+      .selectFrom('ListItem')
+      .select(ListItemSelect)
+      .orderBy('sortOrder', 'asc')
+      .orderBy('createdAt', 'asc')
+      // Temporary tiebreaker sort for deterministic response
```
Comment (bassrock) [inline servers/shareable-lists-api/src/models/ShareableListItem.ts:18]: whats the long term?

File: servers/shareable-lists-api/src/database/OffsetPagination.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * have a consistent experience with our cursor-based
+ * pagination pattern (even when offset is more appropriate)
+ */
+export class OffsetPagination<TB extends keyof DB, O> {
+  public static sentinel = '__pkt__';
+  /**
+   * Create a fake cursor from an offset value
+   */
+  public static encodeCursor(n: number): string {
+    // Probably unnecessary, but give the "cursors" a little oomph
```
Comment (bassrock) [inline servers/shareable-lists-api/src/database/OffsetPagination.ts:18]: 💥

**Review by bassrock — APPROVED** [review]
lgtm! really nice work on the pagination here

File: servers/shareable-lists-api/src/models/ShareableListItem.ts (hunk 2 above)
Comment (kschelonka) [reply servers/shareable-lists-api/src/models/ShareableListItem.ts:18]: either changing the sortOrder field so it doesn't allow collisions if sortOrder isn't explicitly set, or staying with this (or any other numeric ID)

File: servers/shareable-lists-api/src/database/mutations/ShareableList.ts (hunk 1 above)
Comment (kschelonka) [reply servers/shareable-lists-api/src/database/mutations/ShareableList.ts:176]: I know... should I be better? yes. but is a number in javascript going to exceed bigint realistically?

**Pull request merged**